AK/Math.h

```diff
@@ -424,48 +432,70 @@ using Trigonometry::tan;
 namespace Exponentials {
 
 template<FloatingPoint T>
-constexpr T log(T x)
+constexpr T log2(T x)
 {
-    CONSTEXPR_STATE(log, x);
+    CONSTEXPR_STATE(log2, x);
 
 #if ARCH(X86_64)
     T ret;
     asm(
-        "fldln2\n"
+        "fld1\n"
         "fxch %%st(1)\n"
         "fyl2x\n"
         : "=t"(ret)
         : "0"(x));
     return ret;
 #else
 #    if defined(AK_OS_SERENITY)
-    // TODO: Add implementation for this function.
-    TODO();
+    if (x == 0)
+        return Infinity<T>;
+    if (x <= 0 || __builtin_isnan(x))
+        return NaN<T>;
+
+    // FIXME: Make this more accurate, currently it shows errors of up to 0.0017
+    //        If this is done, use it for x86 floats/doubles as well
+    FloatExtractor<T> ext { .d = x };
+
+    FloatExtractor<T> mantissa_ext {
+        .sign = ext.sign,
+        .exponent = FloatExtractor<T>::exponent_bias,
+        .mantissa = ext.mantissa
+    };
+    // (1 <= mantissa < 2)
+    T mantissa = mantissa_ext.d;
+
+    // Approximation using the 4th degree taylor approximation around 1.5
```

Comment:
Oh, interesting, why the choice for 1.5? I would have expected the geometric mean instead of the arithmetic mean, i.e. sqrt(2)=1.414…

Also, how did you measure accuracy?